### PR TITLE
IF: Integration tests enabled for instant-finality (Part 2)

### PR DESCRIPTION
**Pull request by heifner**
Adding additional integration tests for instant-finality

- nodeos_retry_transaction_if_lr_test
- nodeos_run_if_test
- nodeos_short_fork_take_over_if_test
- nodeos_snapshot_diff_if_test (disabled) requires #1558 
- nodeos_snapshot_forked_if_test
- nodeos_under_min_avail_ram_lr_test modified to run only under instant-finality (shouldn't make any difference)
- nodeos_voting_if_lr_test
- p2p_multiple_listen_if_test
- p2p_sync_throttle_if_test
- read-only-trx-basic-if-test
- read-only-trx-parallel-if-test
- read-only-trx-parallel-eos-vm-oc-test modified to run only under instant-finality
- read-only-trx-parallel-no-oc-test modified to run only under instant-finality
- restart-scenarios-if-test-resync
- restart-scenarios-if-test-hard_replay (disabled) requires #2141 
- restart-scenarios-if-test-none
- subjective_billing_test modified to run only under instant-finality
- terminate-scenarios-if-test-resync
- terminate-scenarios-if-test-replay (disabled) requires #2141 
- terminate-scenarios-if-test-hard_replay
- trace_plugin_test modified to run only under instant-finality
- trx_finality_status_forked_if_test
- trx_finality_status_if_test
- ship_streamer_if_test  (disabled) requires #2172

### CODE REVIEW

File: tests/CMakeLists.txt [hunk 1]
```diff
@@ -131,9 +133,14 @@ add_test(NAME ship_test COMMAND tests/ship_test.py -v --num-clients 10 --num-req
 set_property(TEST ship_test PROPERTY LABELS nonparallelizable_tests)
 add_test(NAME ship_test_unix COMMAND tests/ship_test.py -v --num-clients 10 --num-requests 5000 ${UNSHARE} --unix-socket WORKING_DIRECTORY ${CMAKE_BINARY_DIR})
 set_property(TEST ship_test_unix PROPERTY LABELS nonparallelizable_tests)
+add_test(NAME ship_if_test COMMAND tests/ship_test.py -v --activate-if --num-clients 10 --num-requests 5000 ${UNSHARE} WORKING_DIRECTORY ${CMAKE_BINARY_DIR})
```
Comment (linh2931) [inline tests/CMakeLists.txt:136]: why is `--unix-socket` dropped in IF version?

File: tests/CMakeLists.txt [hunk 2]
```diff
@@ -148,9 +155,13 @@ add_test(NAME read-only-trx-basic-test COMMAND tests/read_only_trx_test.py -p 2
...
 set_property(TEST read-only-trx-parallel-test PROPERTY LABELS nonparallelizable_tests)
-add_test(NAME read-only-trx-parallel-eos-vm-oc-test COMMAND tests/read_only_trx_test.py -p 2 -n 3 --eos-vm-oc-enable all --read-only-threads 16 --num-test-runs 3 ${UNSHARE} WORKING_DIRECTORY ${CMAKE_BINARY_DIR})
+add_test(NAME read-only-trx-basic-if-test COMMAND tests/read_only_trx_test.py -p 2 -n 3 --read-only-threads 1 --activate-if ${UNSHARE} WORKING_DIRECTORY ${CMAKE_BINARY_DIR})
+set_property(TEST read-only-trx-basic-if-test PROPERTY LABELS nonparallelizable_tests)
+add_test(NAME read-only-trx-parallel-if-test COMMAND tests/read_only_trx_test.py -p 2 -n 3 --read-only-threads 16 --num-test-runs 3 --activate-if ${UNSHARE} WORKING_DIRECTORY ${CMAKE_BINARY_DIR})
+set_property(TEST read-only-trx-parallel-if-test PROPERTY LABELS nonparallelizable_tests)
+add_test(NAME read-only-trx-parallel-eos-vm-oc-test COMMAND tests/read_only_trx_test.py -p 2 -n 3 --eos-vm-oc-enable all --read-only-threads 16 --num-test-runs 3 --activate-if ${UNSHARE} WORKING_DIRECTORY ${CMAKE_BINARY_DIR})
```
Comment (linh2931) [inline tests/CMakeLists.txt:162]: Do you want to change its name to include `if` as it tests IF mode? Same as `read-only-trx-parallel-no-oc-test`?

Or maybe not, as those two have only one version.

File: tests/CMakeLists.txt (hunk 1 above)
Comment (heifner) [reply tests/CMakeLists.txt:136]: I didn't think an IF version of the test using `unix-socket` would provide any benefit.

**Review by linh2931 — APPROVED** [review]

**Review by greg7mdp — APPROVED** [review]

File: tests/CMakeLists.txt (hunk 2 above)
Comment (linh2931) [reply tests/CMakeLists.txt]: Thanks for  adding `if`. That might help us to quickly find if a test fails in IF or not.

**Comment by ericpassmore** [comment]
Note:start
group: IF
category: TEST
summary: Additional tests covering areas for read-only-trx, snapshots, restarts, and forks among others. Second large testing PR for Faster Finality. 
Note: end

**Pull request merged**